app/controllers/api/base_controller.rb

```diff
@@ -107,6 +107,15 @@ def initialize
     end
 
     before_action :parse_api_request, :log_api_request, :validate_api_request, :validate_api_action
+    before_action :validate_response_format, :only => [:show]
```

Comment:
Any deletes to resources that can potentially take a long time are done async, so providers and vms for example.

They can be either POST's  (i.e. update) via the "delete" action, or DELETE's (i.e. destroy) to the resource.

If we enable the above before_action for all, the only issue that may arise is if scripts send us the DELETE method to let's say /api/providers/:id, and have an Accept in their headers that does not include blank/ */* or */json.  Probably won't be the case since they use the same client for the other methods, but you never know.

What you have is probably safer to have for now.  Thanks!!